### ISSUE TITLE
Use Uint8Array instead of Buffer

### ISSUE THREAD

**Issue by jstarry** [pabigot/buffer-layout#28]
Hey @pabigot, thanks for this library! We have been using it extensively for manual deserialization of blockchain data at Solana. I would like to propose that this library transition to be more browser friendly by using the `Uint8Array` type over the Node.js specific `Buffer` type. Currently we rely on browser polyfills but the latest bundler tools are dropping support for those so it'd be great to remove the need for polyfills entirely.

Are you open to this change being made to the next major release of `buffer-layout`? I'd be happy to own the transition since if we don't upstream this change, I'll be forking this package instead.

**Comment by pabigot** [comment]
Yes, given how many people are using this in non-Node.js environments decoupling it makes sense.  I'm migrating away from Node for the sorts of applications where I needed this library, so I'm open to you providing the changes necessary for the transition, and transitioning ownership of the resulting v2.0.0 to you.  Probably best to start with a PR.

**Comment by jstarry** [comment]
Alright sounds good, will have a PR up soon
